client/src/components/PopularityFilter.tsx

```diff
@@ -0,0 +1,69 @@
+import styles from "./PopularityFilter.module.css";
+import { useState } from "react";
+
+interface PopularityFilterProps {
+  selected: "Unknown" | "Low" | "Medium" | "Any";
```

Comment:
move this type into a file and import it in the components that require it, if you want to, you can check the doc about Enum https://www.typescriptlang.org/docs/handbook/enums.html